examples/shadow_neighborloader.py

```diff
@@ -0,0 +1,94 @@
+import os.path as osp
+
+import torch
+import torch.nn.functional as F
+from tqdm import tqdm
+
+from torch_geometric.datasets import Flickr
+from torch_geometric.loader import NeighborLoader
+from torch_geometric.nn import SAGEConv
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Flickr')
+dataset = Flickr(path)
+data = dataset[0]
+
+kwargs = {'batch_size': 1024, 'num_workers': 10, 'persistent_workers': True}
+
+# Mimic ShadowGNN by setting the argument subgraph_type='induced'.
+# This along with the num_neighbors argument will get the
+# complete induced subgraph
+train_loader = NeighborLoader(data, num_neighbors=[5] * 2,
+                              input_nodes=data.train_mask,
+                              subgraph_type='induced', **kwargs)
+val_loader = NeighborLoader(data, num_neighbors=[5] * 2,
+                            input_nodes=data.val_mask, subgraph_type='induced',
+                            **kwargs)
+test_loader = NeighborLoader(data, num_neighbors=[5] * 2,
+                             input_nodes=data.test_mask,
+                             subgraph_type='induced', **kwargs)
+
+
+class GNN(torch.nn.Module):
+    def __init__(self, in_channels, hidden_channels, out_channels):
+        super().__init__()
+        self.conv1 = SAGEConv(in_channels, hidden_channels)
+        self.conv2 = SAGEConv(hidden_channels, hidden_channels)
+        self.conv3 = SAGEConv(hidden_channels, hidden_channels)
+        self.lin = torch.nn.Linear(hidden_channels, out_channels)
+
+    def forward(self, x, edge_index, batch, root_n_id):
+        x = self.conv1(x, edge_index).relu()
+        x = F.dropout(x, p=0.3)
+        x = self.conv2(x, edge_index).relu()
+        x = F.dropout(x, p=0.3, training=self.training)
+        x = self.conv3(x, edge_index).relu()
+        x = F.dropout(x, p=0.3, training=self.training)
+
+        # We merge both central node embeddings and subgraph embeddings:
+
+        # x = torch.cat([x[root_n_id], global_mean_pool(x, batch)], dim=-1)
```

Comment:
Is that accidentally commented out?

Reply:
Thanks, this line is not needed. I shall delete the commented line.